internal/db/typed.go

```diff
@@ -0,0 +1,140 @@
+// Copyright (C) 2014 The Syncthing Authors.
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+package db
+
+import (
+	"database/sql"
+	"encoding/binary"
+	"errors"
+	"time"
+)
+
+// Typed is a simple key-value store using a specific namespace within a
+// lower level KV.
```

Comment:
Why "typed" though?

(So nice to have my own private review where I can ask all the stupid/unimportant/nitty questions to get the out of my brain without wasting anyones time :) (if you are reading this and are not me, that's your own mistake ;) ))

Reply:
Maybe `Prefixed` instead?